### PR TITLE
feat(kafka): add iam authentication

### PR DESCRIPTION
**Pull request by paul-nicolas**
Fixes ENG-492

### CODE REVIEW

File: components/ledger/libs/aws/iam/load.go [hunk 1]
```diff
@@ -23,6 +24,7 @@ func InitFlags(flags *pflag.FlagSet) {
 	flags.String(AWSSecretAccessKeyFlag, "", "AWS secret access key")
 	flags.String(AWSSessionTokenFlag, "", "AWS session token")
 	flags.String(AWSProfileFlag, "", "AWS profile")
+	flags.String(AWSRoleArnFlag, "", "AWS Role ARN")
```
Comment (Dav-14) [inline components/ledger/libs/aws/iam/load.go:27]: The flag was missing ?

Comment (paul-nicolas) [reply components/ledger/libs/aws/iam/load.go:27]: Yes this is why I added it

Comment (gfyrag) [reply components/ledger/libs/aws/iam/load.go:27]: Flags are not exhaustive, the are a lot of other potential flags.

File: components/operator/internal/resources/auths/controller.go [hunk 2]
```diff
@@ -59,10 +60,20 @@ func Reconcile(ctx Context, stack *v1beta1.Stack, auth *v1beta1.Auth, version st
...
+		if err != nil {
+			return errors.Wrap(err, "creating migrate container")
+		}
+
 		if IsGreaterOrEqual(version, "v2.0.0-rc.5") && databases.GetSavedModuleVersion(database) != version {
 			if err := jobs.Handle(ctx, auth, "migrate",
-				databases.MigrateDatabaseContainer(image, database),
-				jobs.WithServiceAccount(database.Status.URI.Query().Get("awsRole")),
+				migrateContainer,
+				jobs.WithServiceAccount(serviceAccountName),
```
Comment (Dav-14) [inline components/operator/internal/resources/auths/controller.go:76]: retrieving serviceAccountName & migrateContainer is not needed when version < v2.0.0-rc.5"

File: components/operator/internal/resources/settings/aws_role.go [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package settings
+
+import "github.com/formancehq/operator/internal/core"
+
+func GetAWSRole(ctx core.Context, stackName string) (string, error) {
+	return GetStringOrEmpty(ctx, stackName, "awsRole")
```
Comment (gfyrag) [inline components/operator/internal/resources/settings/aws_role.go:6]: Other use this-type-of-case, can you do the same?

File: components/operator/internal/resources/orchestrations/controller.go [hunk 4]
```diff
@@ -61,10 +62,20 @@ func Reconcile(ctx Context, stack *v1beta1.Stack, o *v1beta1.Orchestration, vers
...
+		if err != nil {
+			return errors.Wrap(err, "creating migrate container")
+		}
+
 		if IsGreaterOrEqual(version, "v2.0.0-rc.5") && databases.GetSavedModuleVersion(database) != version {
 			if err := jobs.Handle(ctx, o, "migrate",
-				databases.MigrateDatabaseContainer(image, database),
-				jobs.WithServiceAccount(database.Status.URI.Query().Get("awsRole")),
+				migrateContainer,
+				jobs.WithServiceAccount(serviceAccountName),
```
Comment (Dav-14) [inline components/operator/internal/resources/orchestrations/controller.go:78]: Same as before

**Review by gfyrag — APPROVED** [review]

File: components/operator/internal/resources/payments/controller.go [hunk 5]
```diff
@@ -61,15 +63,25 @@ func Reconcile(ctx Context, stack *v1beta1.Stack, p *v1beta1.Payments, version s
...
-				databases.MigrateDatabaseContainer(image, database,
-					func(m *databases.MigrationConfiguration) {
-						m.AdditionalEnv = []corev1.EnvVar{
-							Env("CONFIG_ENCRYPTION_KEY", encryptionKey),
-						}
-					},
-				),
-				jobs.WithServiceAccount(database.Status.URI.Query().Get("awsRole")),
+				migrateContainer,
+				jobs.WithServiceAccount(serviceAccountName),
```
[the remark this answers is not in the document]
Comment (paul-nicolas) [reply components/operator/internal/resources/payments/controller.go:84]: We didn't have any version check before, I think not

**Pull request merged**